rules/network/command_and_control_fin7_c2_behavior.toml

```
@@ -0,0 +1,54 @@
+[metadata]
+creation_date = "2020/07/06"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/07/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects a known command and control pattern in network events. The FIN7 threat group is known to use this
+command and control technique, while maintaining persistence in their target's network.
+"""
+false_positives = [
+    """
+    This rule could identify benign domains that are formatted similarly to FIN7's command and control algorithm. Alerts
+    should be investigated by an analyst to assess the validity of the individual observations.
+    """,
+]
+index = ["packetbeat-*"]
+language = "lucene"
+license = "Elastic License"
+name = "Possible FIN7 Command and Control Behavior"
+note = "This activity has been observed in FIN7 campaigns."
+references = [
+    "https://www.fireeye.com/blog/threat-research/2018/08/fin7-pursuing-an-enigmatic-and-evasive-global-criminal-operation.html",
+]
+risk_score = 73
+rule_id = "4a4e23cf-78a2-449c-bac3-701924c269d3"
+severity = "high"
+tags = ["Elastic", "Network"]
+type = "query"
+
+query = '''
+event.category:(network OR network_traffic) AND type:(tls OR http) AND network.transport:tcp AND destination.domain:/[a-zA-Z]{4,5}\.(pw|us|club|info|site|top)/ NOT destination.domain:zoom.us
```

missing an AND. I don't remember whether AND or OR is implicit for lucene, but we should make it explicit
```suggestion
event.category:(network OR network_traffic) AND type:(tls OR http) AND network.transport:tcp
AND destination.domain:/[a-zA-Z]{4,5}\.(pw|us|club|info|site|top)/ AND NOT destination.domain:zoom.us
```

---

I like the suggestion to add DGA to the name for cases when this FPs - since it could make triage more obvious

---

FWIW, if not otherwise specified, multiple terms in a Lucene query default to `AND` (e.g. `category: network type: tls`) and multiple terms within a fielded search, default to `OR` (e.g. `type: (tls http)`). `NOT` is just a negation and follows these existing rules. To be fair of the Lucene critics out there, it's actually more complicated than that because `AND`, `OR`, and `NOT` merely express a _should_ relationship by changing the scoring of a document, whereas the `+` and `-` operator specify a _must_ relationship which must evaluate true to be returned.

:themoreyouknow: